src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -352,18 +379,21 @@ function getSourceForTree(state: AppState, source: ?Source): ?Source | null {
   return getGeneratedSourceByURL(state, getRawSourceURL(source.url));
 }
 
-const mapStateToProps = state => {
+const mapStateToProps = (state, props) => {
   const selectedSource = getSelectedSource(state);
   const shownSource = getShownSource(state);
+  const focused = getFocusedSourceItem(state);
 
   return {
     shownSource: getSourceForTree(state, shownSource),
     selectedSource: getSourceForTree(state, selectedSource),
     debuggeeUrl: getDebuggeeUrl(state),
-    expanded: getExpandedState(state),
+    expanded: getExpandedState(state, props.thread),
+    focused: focused && focused.thread == props.thread ? focused.item : null,
     projectRoot: getProjectDirectoryRoot(state),
     sources: getRelativeSources(state),
```

Comment:
it might be nice to have a selector `getRelativeSourcesForThread` this will simplify some of the logic in the tree and perhaps keep the tree from updating if a source for another thread is received.